### PR TITLE
Remove solid_mechanics

### PR DESCRIPTION
**Pull request by bwspenc**
ref #15575

### CODE REVIEW

**Comment by bwspenc** [comment]
I know this still needs some work, and will fail some of the modules tests. I'm mostly interested in seeing at this point which of the downstream apps fail.

**Comment by bwspenc** [comment]
FYI @jasondhales @dschwen @novasr

File: modules/xfem/doc/config.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ Extensions:
         executable: ${MOOSE_DIR}/modules/xfem
         hide:
             framework: !include framework/doc/hidden.yml
-            xfem: !include modules/xfem/doc/hidden.yml
+            #xfem: !include modules/xfem/doc/hidden.yml
```
Comment (bwspenc) [inline modules/xfem/doc/config.yml:18]: Can you just include this anyway even though it's empty?

Comment (vanwdani) [reply modules/xfem/doc/config.yml:18]: I followed the convention set forth by geochemistry.  It gave me an error previously on this

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**